frontend/src/lib/components/DurationPicker/DurationPicker.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import './DurationPicker.scss'
+import { SmallTimeUnit } from '~/types'
+import { durationPickerLogic } from './durationPickerLogic'
+import { useActions, useValues } from 'kea'
+import { Input, Select } from 'antd'
+import { capitalizeFirstLetter } from 'lib/utils'
+
+interface Props {
+    onChange: (value: number) => void
+    key: string
+    initialValue?: number
+    autoFocus?: boolean
+    style?: Partial<React.CSSProperties>
+}
+
+export const durationOptions: SmallTimeUnit[] = ['seconds', 'minutes', 'hours']
+
+export function DurationPicker({ initialValue, onChange, key, autoFocus, style }: Props): JSX.Element {
```

Comment:
From my understanding, `key` here will get swallowed by React's internal piping and will always be undefined.

Reply:
Ooooo good catch!